src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1544,6 +1581,9 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     else
     {
       assert(cl.static_lifetime);
+      if(t.is_vector())
+        // Global vectors are compound operands that need to be early initialized
+        new_expr.swap(initializer);
```

Comment:
For example:
```c
v4si vsi = (v4si){1, 2, 3, 4}; // global int vector
```

Will be parsed as a compound literal, resulting in:
```
vsi=main.c:7$compound-literal$1;
main.c:7$compound-literal$1={ 1, 2, 3, 4 };
```

However, vectors are not references. By doing the early initialization, it will initialize as:
```
vsi={ 1, 2, 3, 4 };
main.c:7$compound-literal$1={ 1, 2, 3, 4 };
```

I am not sure whether I should remove the compound-literal though.

Reply:
Oh, I see, it's still initialized in the wrong order. I thought we fixed that. I can take a look.

> I am not sure whether I should remove the compound-literal though.
I don't think you can; they're really anonymous symbols that can be manipulated through pointers or assignments:
```c
	v4si *p = &(v4si){1,2,3,4};
	(*p)[3] = 23;            /* well-defined */
	printf("%d\n", (*p)[3]); /* prints 23 */
```

Reply:
I've put a possible fix for this in #816. Let's wait for the CI to confirm this.

Reply:
#816 fixed the issue, I reverted the changes